programs/server/users.xml

```diff
@@ -27,6 +27,38 @@
         </readonly>
     </profiles>
 
+    <!-- External user directories (LDAP). -->
+    <ldap_servers>
+        <!-- List LDAP servers with their connection parameters here to later use them as authenticators for dedicated users,
+             who have 'ldap' authentication mechanism specified instead of 'password'.
+             Parameters:
+                host - LDAP server hostname or IP, this parameter is mandatory and cannot be empty.
+                port - LDAP server port, default is 636 if enable_tls is set to true, 389 otherwise.
+                auth_dn_prefix, auth_dn_suffix - prefix and suffix used to construct the DN to bind to.
+                        Effectively, the resulting DN will be constructed as auth_dn_prefix + escape(user_name) + auth_dn_suffix string.
+                        Note, that this implies that auth_dn_suffix should usually have comma ',' as its first non-space character.
+                enable_tls - flag to trigger use of secure connection to the LDAP server.
+                        Specify 'no' for plain text (ldap://) protocol (not recommended).
+                        Specify 'yes' for LDAP over SSL/TLS (ldaps://) protocol (recommended, the default).
+                        Specify 'starttls' for legacy StartTLS protocol (plain text (ldap://) protocol, upgraded to TLS).
+                tls_cert_verify - TLS peer certificate verification behavior.
+                        Accepted values are: never, allow, try, demand.
+                ca_cert_dir - path to CA certificates directory.
+                ca_cert_file - path to CA certificate file.
+             Example: 
+                <my_ldap_server>
+                    <host>localhost</host>
+                    <port>636</port>
+                    <auth_dn_prefix>cn=</auth_dn_prefix>
+                    <auth_dn_suffix>, ou=users, dc=example, dc=com</auth_dn_suffix>
+                    <enable_tls>yes</enable_tls>
+                    <tls_cert_verify>demand</tls_cert_verify>
+                    <ca_cert_dir>/path/to/ca_cert_dir</ca_cert_dir>
```

Comment:
Are default ca used? Do we need to list default ca path explicitly? Is there a way to use self-signed certs?

Reply:
Yes, tls_cert_verify controls that. But the description of the behavior is too long to put it here. Meanwhile, it is a typical config for any OpenSSL powered functionality.

Reply:
And it's a typical place where people tend to have problems. :) those questions are rather  for QA / documentation.

Reply:
May be some settings (like paths) should be inherited by default from global clickhouse ssl client config
https://github.com/ClickHouse/ClickHouse/blob/9d5b1b69f7c3d6279e09ab593c22f6b1ec3ec7bf/programs/server/config.xml#L69

Some of other properties may be also need to be configurable 
https://www.man7.org/linux/man-pages/man3/ldap_get_option.3.html#TLS_OPTIONS